napari/layers/_layer_actions.py

```diff
@@ -50,13 +50,16 @@ def _split_rgb(ll: LayerList):
 def _convert(ll: LayerList, type_: str):
     from napari.layers import Shapes
 
+    float_dtypes = [np.float64, np.float32, np.float16]
     for lay in list(ll.selection):
         idx = ll.index(lay)
         ll.pop(idx)
         if isinstance(lay, Shapes) and type_ == 'labels':
             data = lay.to_labels()
+        elif lay.data.dtype in float_dtypes and type_ == 'labels':
+            data = lay.data.astype(int)
```

Comment:
did we want to always cast to `int` Not `uint` or event `uint16` 

I know that it requires to calculate `min` and `max` which requires additional CPU time but will allow reduced memory usage.

Reply:
But calculating the min & max for very large datasets is often quite a time consuming operation. I do not think that should be the default.

Reply:
It is not expansive in the context of casting a whole array from float to int. There is also `contrast_limits` property of the image layer.

Reply:
I think it should be int all the time. Converting to lesser types, even if the min/max fit, is problematic, because it limits how the user can edit the labels.

At any rate, even if we do want that optimisation, it can come in a later PR.